### PR TITLE
MINIFICPP-411: Resolve issues with command reporting. Show the size b…

### PR DESCRIPTION
**Pull request by phrocker**
…efore clear, and fix docs to clarify that a response isn't immediate

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: controller/MiNiFiController.cpp [hunk 1]
```diff
@@ -178,7 +177,7 @@ int main(int argc, char **argv) {
       auto& components = result["stop"].as<std::vector<std::string>>();
       for (const auto& component : components) {
         auto socket = secure_context != nullptr ? stream_factory_->createSecureSocket(host, port, secure_context) : stream_factory_->createSocket(host, port);
-        if (!stopComponent(std::move(socket), component))
+        if (stopComponent(std::move(socket), component))
```
Comment (apiri) [inline controller/MiNiFiController.cpp]: I agree this is definitely the right intent, but it appears as though we are not handling/propagating the returned value for writeData in sendSingleCommand which always returns true.

File: README.md [hunk 2]
```diff
@@ -712,6 +712,7 @@ These are defined by default to the above values. If the port option is left und
 will be disabled in your deployment.
 
  The executable is stored in the bin directory and is titled minificontroller. Available commands are listed below.
+ Note that with all commands and immediate response by the agent isn't guaranteed. In all cases the agent assumes the role of validating that a response was received, but execution of said command may take some time depending on a number of factors to include persistent storage type, size of queues, and speed of hardware.
```
Comment (apiri) [inline README.md]: minor: and -> an

File: controller/MiNiFiController.cpp (hunk 1 above)
Comment (phrocker) [reply controller/MiNiFiController.cpp]: The intent was to actually return false when writes failed. This is an oversight. Thanks

**Comment by phrocker** [comment]
Here is an example response @apiri when it is not running. Is this not what you see? 
[HOST:bin phrocker$  ./minificontroller --stop GetFile
Could not connect to remote host localhost:9998

**Comment by apiri** [comment]
Hey @phrocker it was not, but it looks like the changes you added in 645e355 would resolve the hardcoded true.  Not sure of exact sequence of your reply to my initial review and this last comment as it was edited.  The new changes look to resolve the issue.

**Comment by apiri** [comment]
@phrocker looks good.  will merge

**Pull request closed without merging**